### PR TITLE
Add implementation for shadows of AccessibilityManagers StateChangeListener

### PR DESCRIPTION
**Pull request by MichaelEvans**
### Overview
Previously this shadow just returned `true`, now providing an implementation similar to the one used in `ConnectivityManager#addDefaultNetworkActiveListener`.

### Proposed Changes
Added a simple implementation as well as a handful of tests.

### CODE REVIEW

File: robolectric/src/test/java/org/robolectric/shadows/ShadowAccessibilityManagerTest.java
Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowAccessibilityManagerTest.java:153]: What about adding `getEnabled` method to custom listener, and using it instead of `Mockito#spy` to check state changed?

Comment (MichaelEvans) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowAccessibilityManagerTest.java:153]: That works for me - I saw usages of `spy` elsewhere so I thought I'd try to follow conventions, I much prefer your suggestion. Will update this PR!

File: robolectric/src/test/java/org/robolectric/shadows/ShadowAccessibilityManagerTest.java
Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowAccessibilityManagerTest.java:178]: Make it "private static"? Looks like we don't need open it outside of this test class.

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowAccessibilityManager.java
Comment (utzcoz) [inline shadows/framework/src/main/java/org/robolectric/shadows/ShadowAccessibilityManager.java:116]: Looks like https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/view/accessibility/AccessibilityManager.java;l=852-859?q=removeAccessibilityStateChangeListener&ss=android, https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/util/ArrayMap.java;l=725-732;drc=master, https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/util/ArrayMap.java;l=433-436;drc=master will not throw `IllegalArgumentException` if listener is null.

Comment (MichaelEvans) [reply shadows/framework/src/main/java/org/robolectric/shadows/ShadowAccessibilityManager.java]: Good catch, fixed!

**Review by utzcoz — APPROVED** [review]
@MichaelEvans Thanks. LGTM.

**Comment by MichaelEvans** [comment]
@utzcoz Thank you for the review, I appreciate the eyes on it!

**Comment by utzcoz** [comment]
> @utzcoz Thank you for the review, I appreciate the eyes on it!

Thanks for your contributing. Please wait for @hoisie confirmation that it doesn't break Google internal test suites. I think he will merge it if there is no extra issue.

**Comment by utzcoz** [comment]
Hi @hoisie , could you help to review this PR?

**Comment by utzcoz** [comment]
It's better if we can clear cached state/listener by `List` at `resetter` method. But we can do it at later PRs, because old cached state/listener doesn't be cleared too.

**Comment by hoisie** [comment]
Thanks a lot for the contribution! I am validating this PR against some tests internally

**Review by hoisie — APPROVED** [review]

**Comment by MichaelEvans** [comment]
@hoisie Not a problem at all, thank you! 😄

**Comment by utzcoz** [comment]
> It's better if we can clear cached state/listener by `List` at `resetter` method. But we can do it at later PRs, because old cached state/listener doesn't be cleared too.

It's my mistake. The `resetter` will reset singleton instance of `ShadowAccessibilityManager`, and new instance will get new internal state.

**Pull request merged**
